src/domain/ledger/index.types.d.ts

```diff
@@ -20,14 +20,14 @@ type LedgerJournal = {
 }
 
 // Differentiate fields depending on what 'type' we have (see domain/wallets/index.types.d.ts)
-type LedgerTransaction = {
+type LedgerTransaction<S extends WalletCurrency> = {
   readonly id: LedgerTransactionId
   readonly walletId: WalletId | undefined // FIXME create a subclass so that this field is always set for liabilities wallets
   readonly type: LedgerTransactionType
-  readonly debit: Satoshis
-  readonly credit: Satoshis
+  readonly debit: S extends "BTC" ? Satoshis : UsdCents
```

Comment:
We want to move away from these types completely - I guess you are looking to do this in another PR?

Reply:
Yes, didn't want to refactor all the other places outside payments that `LedgerTransaction` type is also used quite as yet